### DISCUSSION TITLE
Doubts about Rate Group

### DISCUSSION THREAD

**Discussion by sheershak-agarwal** [nasa/fprime#1052] [Q&A]
High Level: I am using GPS pps to correct the unix time and using Rate groups in my other components to do periodic check, between separate components which only has gps time in common.

**is there a risk that the two separate clocks will not be synced or will there be a clock drift?**

I know the path for the Active Rate Group is 
Svc::TimerVal(Unix Time)->BlockDriver->RateGroupComponent->ActiveRateGroup

**Comment by LeStarch** [comment]
Are you asking about a drift between the clock that starts a rategroup and the PPS?  or the time supplied to a rategroup

**Reply by sheershak-agarwal** [reply]
the time supplied to a rategroup or the clock that starts a rategroup

**Reply by LeStarch** [reply]
The clock that drives the rategroup was intended as an example.  Typically a project will replace BlockDriver with some hardware driver to run the rategroups.  For example, you could use the PPS signal of the GPS in place of the clock.  The clock that drives it now is a "sleep for 1sec, then drive" loop....so it isn't very accurate.

The clock time passed to the rategroup is an a timer offset that traces to `Os::IntervalTimer` on linux this uses the system time: `clock_gettime(CLOCK_REALTIME)`.  Thus the clock should be consistent if your adjustments affect the system CLOCK_REALTIME.
